lib/auth/methods.go

```diff
@@ -371,6 +371,11 @@ func (a *Server) authenticateUserInternal(ctx context.Context, req AuthenticateU
 		authErr = authenticateHeadlessError
 	case req.Webauthn != nil:
 		authenticateFn = func() (*types.MFADevice, error) {
+			if req.Pass != nil {
```

Comment:
@jentfoo Regarding your previous comment about being defensive and returning an error if `Pass` is `nil`: we can't do it. In the MFA login flow, the Web UI first calls `/v1/webapi/mfa/login/begin` with username and password and they get verified while creating the MFA challenge. The MFA challenge response (`/v1/webapi/mfa/login/finishsession`) doesn't get the password in the request, and the `Pass` field is `nil`.

This situation is different from the password change scenario, since in the password change, we are talking about generating any challenge as an already signed-in user (a session hijacking scenario). In this scenario, there's no session to hijack, and the user can't create a challenge without giving a password (unless this is a passwordless challenge, but this code branch doesn't handle the passwordless case).